pkgs/development/tools/sqlelf/default.nix

```
@@ -0,0 +1,68 @@
+{ lib
+, stdenv
+, buildPythonPackage
+, fetchPypi
+
+# build-system
+, setuptools-scm
+
+# dependencies
+, capstone
+, lief
+, apsw
+, sh
+
+# tests
+, pytestCheckHook
+, coreutils
+}:
+
+buildPythonPackage rec {
```

If this is an application, and not a library, it should use

```suggestion
python3Packages.buildPythonApplication rec {
```

in which case all inputs can use `with python3Packages;` to refer to packages within the python package set, and do not need to be referenced as function arguments.

---

Oh, I didn't know about this convention, I'll move it to `by-name` too then

---

If put under the `by-name` directory, I don't think this package will come under the `python3xPackages` namespace anymore. Please confirm this @mweinelt, as I'm doubtful of this too.

---

Only libraries are provided in the `python3Packages` namespace. Application go into top-level, so by-name is fine. I missed the `python3Packages.callPackage` in `all-packages.nix`, which explains the original argument list. Moving to `by-name` is the correct thing to do.

---

Ah, right; I came in with the bias that this was also a library. Tunnel visioned there, haha; sorry.